redis/src/cluster_async/mod.rs

```diff
@@ -1010,9 +1230,17 @@ where
             })
     }
 
+    fn check_if_all_slots_covered(slot_map: &SlotMap) -> bool {
+        let mut slots_covered = 0;
+        for (end, slots) in slot_map.slots.iter() {
+            slots_covered += end - slots.start + 1;
```

Comment:
Its common practice to use `saturating_add` / `saturating_sub` instead of calling `+` or `-`. Or use `checked_add` / `checked_sub` to protect from overflows